tests/sentry/api/endpoints/test_auth_index.py

```diff
@@ -164,374 +168,340 @@ def test_invalid_password_u2f(self, validate_response, is_available, mock_metric
+                    "superuserAccessCategory": "for_unit_test",
+                    "superuserReason": "for testing",
+                },
+                SERVER_NAME=f"{other_org.slug}.testserver",
+                HTTP_REFERER=referrer,
+            )
+            # status code of 401 means invalid SSO session
+            assert response.status_code == 401
+            query_string = urlencode({"next": referrer})
+            assert response.data == {
+                "detail": {
+                    "code": "sso-required",
+                    "extra": {
+                        "loginUrl": f"http://{self.organization.slug}.testserver/auth/login/{self.organization.slug}/?{query_string}"
+                    },
+                    "message": "Must login via SSO",
+                }
+            }
+            assert COOKIE_NAME not in response.cookies
+
+    @override_settings(SENTRY_SELF_HOSTED=False)
     @with_feature("organizations:u2f-superuser-form")
     def test_superuser_sso_user_no_u2f_saas_product(self):
-        from sentry.auth.superuser import Superuser
-
-        with self.settings(SENTRY_SELF_HOSTED=False):
-            org_provider = AuthProvider.objects.create(
-                organization_id=self.organization.id, provider="dummy"
-            )
+        org_provider = AuthProvider.objects.create(
+            organization_id=self.organization.id, provider="dummy"
+        )
 
-            user = self.create_user("foo@example.com", is_superuser=True)
+        user = self.create_user("foo@example.com", is_superuser=True)
 
-            create_authenticator(user)
+        create_authenticator(user)
 
-            AuthIdentity.objects.create(user=user, auth_provider=org_provider)
+        AuthIdentity.objects.create(user=user, auth_provider=org_provider)
 
-            with mock.patch.object(Superuser, "org_id", self.organization.id):
-                with self.settings(SENTRY_SELF_HOSTED=False):
-                    self.login_as(user, organization_id=self.organization.id)
-                    response = self.client.put(
-                        self.path,
-                        data={
-                            "isSuperuserModal": True,
-                            "superuserReason": "for testing",
-                        },
-                    )
-                    assert response.status_code == 403
+        with mock.patch.object(Superuser, "org_id", self.organization.id):
+            self.login_as(user, organization_id=self.organization.id)
+            response = self.client.put(
+                self.path,
+                data={
+                    "isSuperuserModal": True,
+                    "superuserReason": "for testing",
+                },
+            )
+            assert response.status_code == 403
+            assert response.data["detail"]["code"] == "ignore"
 
+    @override_settings(SENTRY_SELF_HOSTED=False, VALIDATE_SUPERUSER_ACCESS_CATEGORY_AND_REASON=True)
```

Comment:
`VALIDATE_SUPERUSER_ACCESS_CATEGORY_AND_REASON` is already `True` in sentry? https://github.com/getsentry/sentry/blob/a9895178b56b9cf20e4b4d18d09ff0b005ec230d/src/sentry/conf/server.py#L3795

Reply:
ah youre right, got rid of that too